### ISSUE TITLE
QB Binding mismatch: Hasmany .exists() or .count() replacing foreignKey with "true"

### ISSUE THREAD

**Issue by MordantWastrel** [coldbox-modules/quick#112]
(4.0.2)

We have `invoiceInstance` with the following `hasMany`:

```
function subscriptions() {
	return hasMany(
		relationName = "qInvoiceSubscription@invoices",
		foreignKey = "instanceID",
		localKey = "instanceID"
	);
}
```

The relationship works correctly if we call `getSubscriptions()`.

If we call `subscriptions().exists()` or `subscriptions().count()` on, say, instanceID 46, the query that runs is looks for an instanceID of `true` rather than `46`.

```
SELECT COUNT(*) AS "aggregate" FROM [invoice_subscriptions] 
WHERE ([invoice_subscriptions].[instanceID] = true 
AND [invoice_subscriptions].[instanceID] IS NOT NULL) AND [invoice_subscriptions].[instanceID] IS NOT NULL)
```

The `sqltype` for `instanceID` is `integer` on both entities. This worked in v3.

**Comment by MordantWastrel** [comment]
This isn't occurring on every hasMany so there's something else going on here but I couldn't get into it over the weekend. I'll get some more information this morning and update this ticket.

**Comment by MordantWastrel** [comment]
The problem appears to be that there is an extra binding being generated that is taking the place of the correct binding. 

In `QueryBuilder.cfc`, the query is correct (coming from the Quick entity) until it hits `withAggregate()` inside of `aggregateQuery()`, at which point `toSQL()` produces this:

```
SELECT COUNT(*) AS "aggregate" FROM [invoice_subscriptions] WHERE ([invoice_subscriptions].[instanceID] = {"value":true,"cfsqltype":"bit","null":false} AND [invoice_subscriptions].[instanceID] IS NOT NULL) 
```

Getbindings() shows that it thinks the first argument's `cfsqltype` is `bit`.  But `getWheres()` looks correct to me assuming that `basic` is the correct type for that numeric value:

![Screenshot 2020-08-17 084157](https://user-images.githubusercontent.com/11874518/90402754-88d1f000-e065-11ea-9688-188f3cd12e4e.png)

That's dumped out right before the query is executed in `get()`.

But `getBindings()` in the same spot shows two bindings instead of one,

![Screenshot 2020-08-17 090650](https://user-images.githubusercontent.com/11874518/90405206-1531e200-e069-11ea-844b-76dcd1de414d.png)

That first binding is shown as a `select` binding if I dump out `bindings`, though. That's as far as I've gotten: Is the issue here that aggregate queries should eliminate SELECT bindings that might be coming from somewhere else? (I don't even know where that one is coming from.)

**Comment by elpete** [comment]
Fixed in qb@8.0.3
